test/functional/rpc_generateblock.py

```diff
@@ -5,33 +5,70 @@
 '''Test generateblock rpc.
 '''
 
+from decimal import Decimal
 from test_framework.test_framework import BitcoinTestFramework
+from test_framework.address import ADDRESS_BCRT1_P2WSH_OP_TRUE
 from test_framework.util import (
     assert_equal,
     assert_raises_rpc_error,
 )
+from test_framework.messages import (
+    COIN,
+    COutPoint,
+    CTransaction,
+    CTxIn,
+    CTxInWitness,
+    CTxOut,
+)
+from test_framework.script import (
+    CScript,
+    OP_TRUE,
+)
+from test_framework.util import (
+    satoshi_round,
+    hex_str_to_bytes,
+)
+from test_framework.wallet import MiniWallet
+
+def create_miniwallet_rawtx(utxo, scriptPubKey):
+        flat_fee = Decimal('0.001')
+        send_value = satoshi_round(utxo['value'] - flat_fee)
+
+        tx = CTransaction()
+        tx.vin = [CTxIn(COutPoint(int(utxo['txid'], 16), utxo['vout']))]
+        tx.vout = [CTxOut(int(send_value * COIN), scriptPubKey)]
+        tx.wit.vtxinwit = [CTxInWitness()]
+        tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
+        rawtx = tx.serialize().hex()
+        return rawtx
 
 
 class GenerateBlockTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        address = ADDRESS_BCRT1_P2WSH_OP_TRUE
+        scriptPubKey = hex_str_to_bytes(node.validateaddress(ADDRESS_BCRT1_P2WSH_OP_TRUE)['scriptPubKey'])
```

Comment:
I know it's frowned upon, but thoughts on doing:
```python
address = miniwallet._address
scriptPubKey = miniwallet._scriptPubKey
```

Otherwise we're just duplicating that logic, and we can get rid of a bunch of imports. Maybe it's worth considering adding a getter for `address` and `scriptPubKey` to `MiniWallet` (although I don't see anything wrong just accessing the private vars)